aeppl/mixture.py

```diff
@@ -300,6 +302,67 @@ def mixture_replace(fgraph, node):
     return [new_mixture_rv]
 
 
+@local_optimizer((Elemwise,))
+def switch_mixture_replace(fgraph, node):
+    r"""Identify mixture sub-graphs defined via a Switch Op and replace them with
+    a place-holder MixtureRV Op
+    """
+    try:
+        if not isinstance(node.op.scalar_op, Switch):
+            return None  # pragma: no cover
+    except AttributeError:
+        return None
+
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    old_mixture_rv = node.default_output()
+
+    if len(node.inputs) != 3:
+        return None  # graph structure should be [index, component1, component2]
+
+    mixing_indices = node.inputs[0]
+    mixture_res = node.inputs[1:]
+
+    # We loop through mixture components and collect all the array elements
+    # that belong to each one (by way of their indices).
+    mixture_rvs = []
+    for i, component_rv in enumerate(mixture_res):
+
+        # We create custom types for the mixture components and assign them
+        # null get_measurable_outputs dispatches so that they aren't
+        # erroneously encountered in places like factorized_joint_logprob.
+        new_node = assign_custom_measurable_outputs(component_rv.owner)
+        out_idx = component_rv.owner.outputs.index(component_rv)
+        new_comp_rv = new_node.outputs[out_idx]
+        mixture_rvs.append(new_comp_rv)
+
+    # Replace this sub-graph with a MixtureRV
+    mix_op = MixtureRV(
+        2,
+        old_mixture_rv.dtype,
+        old_mixture_rv.broadcastable,
+    )
+    new_node = mix_op.make_node(*([NoneConst, mixing_indices] + mixture_rvs))
```

Comment:
I believe that the lines here at the most important.

- 2 as first argument to `MixtureRV` because I'm assuming that `mixing_indices` is just a scalar. I imagine that this will be extended to `1 + len(mixing_indices)` as in `mixture_replace`;
- I put `NoneConst` as the first input to `mix_op.make_node` as I am not sure what axis does `at.switch` work with.